src/with-urql-client.tsx

```diff
@@ -22,12 +22,18 @@ interface WithUrqlState {
   urqlState: SSRData;
 }
 
-interface NextContextWithAppTree extends NextContext {
+export interface NextContextWithAppTree extends NextContext {
   AppTree: React.ComponentType<any>;
 }
 
+type NextUrqlClientOptions =
+  | Omit<ClientOptions, 'exchanges' | 'suspense'>
+  | ((
+      ctx: NextContext<any, any>,
+    ) => Omit<ClientOptions, 'exchanges' | 'suspense'>);
+
```

Comment:
Allowing a user to either pass in the client options directly or define a function that takes in `ctx` and returns `clientOptions`.